### DISCUSSION TITLE
Not working with Quectel_M35

### DISCUSSION THREAD

**Discussion by toandd-it** [IchthysMaranatha/asterisk-chan-quectel#15] [General]
I am using a GSM device Quectel M35, and i am getting error message when using your project.
Please help me debug.
IMG: https://pbx.demo.zetadmin.com/m35/1.PNG

Before that, I edited the AT commands of the chan_dongle project.
Display the complete list of devices. But I haven't been able to make any calls yet.
IMG: https://pbx.demo.zetadmin.com/m35/2.PNG

Thank!

**Comment by IchthysMaranatha** [comment]
If m35 does not support at+dsci=1 command it will not be able to to handle calls with chan_quectel.
Please use minicom and let us know the results of 
at+dsci=1

And there's no way it will ever work with chan_dongle for calls, it does not support Huawei call URCs

**Reply by toandd-it** [reply]
> If m35 does not support at+dsci=1 command it will not be able to to handle calls with chan_quectel. Please use minicom and let us know the results of at+dsci=1
> 
> And there's no way it will ever work with chan_dongle for calls, it does not support Huawei call URCs

m35 does not support at+dsci=1, when using minicom show "ERROR"

**Reply by IchthysMaranatha** [reply]
Sorry, it should have been at^dsci=1
It's moot anyway as you must have understood.

**Comment by IchthysMaranatha** [comment]
To get a modem to work for calls, it must provide some URC at all stages of a call, eg when originating, when connected and when disconnected. If there are no such URCs we cannot modify this code to get it to work. Please go through readme for more details.

**Comment by IchthysMaranatha** [comment]
Looked at the M35 manual and looks like you can AT+QEXTUNSOL to generate needed URCs. You will have to modify code accordingly. 

But does m35 support voice over usb even if all this is done? I don't think it does. And if it does not, everything else is pointless.
